src/Error/Error.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{-|
+Description : The top-level error data types. Specific error types are nested within these.
+-}
+
+module Error.Error where
+
+import Error.TypeError
+import Data.Aeson hiding (Error)
+
+import Text.Parsec.Pos
+
+-- | A top-level error data type that contains information that all errors should have
+data Error = Error {
+                     err  :: Err          -- ^ more specific error information
+                   , eid  :: Int          -- ^ error id; for future use
+                   , epos :: SourcePos    -- ^ error position
+                   }
+   deriving (Eq)
+
+-- Note: the error code functionality works, but only for type errors
+-- the code below can be uncommented when it is introduced holistically
+instance Show Error where
+   show (Error (TE e) i p) = {-ec ++ "TE" ++ show i ++ "\n" ++-} "Type error in " ++ name ++ show e
```

Comment:
Note that this is intentionally commented out.

Reply:
~~Where does `name` come from?~~ nvm, didn't read the where